### ISSUE TITLE
Flaky sync that needs a nudge to work.

### ISSUE THREAD

**Issue by cyphase** [laurent22/joplin#5828]
### Desktop

Joplin version: 2.6.2
Platform: Linux 64-bit
OS specifics: Ubuntu

### Android

Joplin version: 2.6.3
Platform: Android 11 64-bit
OS specifics: LineageOS

## Steps to reproduce

I'd been using filesystem sync for a long time, powered by Syncthing. I recently update to 2.6.3 on Android and hit #5779. To work around the issue, I changed my sync target to WebDAV. First I changed it on the desktop app and synced. Then I changed it on the Android app and synced.

Then I synced again on the desktop. I knew I had some mobile notes that hadn't synced yet since the #5779 issue. But what I didn't expect was to see old notes pop up that should have synced months ago, but never had. I'm not sure if this is related to the following or a separate issue.

Then I noticed that some recent reorganization of notebooks that I'd done on the desktop app since Android filesystem sync broke had not been fully mirrored. Also, the contents of the notebooks were partially updated, but not completely. Some notebooks that had their contents updated on mobile where nonetheless in their old location with their old name. Some notes that I'd moved to other places were still in the old place on mobile. Some notes that I'd recently changed the contents of on desktop were not yet changed on mobile.

I've found that moving a note or notebook somewhere else then back on desktop, then syncing, seems to update something that gets the changes that were already there to now sync to mobile.

I haven't had time to reproduce this from scratch yet, especially since it might have to do with starting with older versions and then updating; but I'm willing to try to do so. I just want to report this sooner rather than later. Perhaps it'll be easy for you to figure out why this is happening.

## Describe what you expected to happen

I expected everything to sync correctly, including locations of notes and notebooks and changes to notes.

## Logfile

This'll need to wait for reproduction, since I already manually fixed the issues (by moving things away and back). If I find something I missed, I'll add a log.

## Possible fix

Something that is being updated when a note or notebook is moved somewhere and moved back, is triggering the sync to work correctly after that point. Perhaps that can be applied in such a way as to make it always work. Or perhaps it's just a sign of what the underlying issue is that needs to be fixed.

---

I am available for any questions. As I said, I wanted to get this down. I'll come back to this with more information when I can, or to respond to any comments.

**Comment by bensoibj** [comment]
If I understand the steps to reproduce correctly, I observed the same behaviour. My observation is the following:

Sometimes the change of a note on one device somehow “sticks”. Even though the change is obviously synced on the filesystem (what can be verified by checking the according file in the filesystem sync folder), the change is not displayed on a second device, there still the note with the older modification time is displayed, even after manually triggering a sync several times. Only after changing the note on the first device again (“nudging” it), the change “comes through”. 

Interestingly, if one changes the old note on the second device without first ensuring that the change from the first device has fully arrived on the second one (not only file-based, but in Joplin), a conflict occurs. The change from the fist device is then properly displayed and the modified note from the second device appears as a conflicting note.

Maybe this is related to #5582, yet there the note is lost with no conflict.
